src/box/iproto.cc

```diff
@@ -2978,6 +2977,16 @@ iproto_send_listen_msg(struct evio_service *binary)
 	struct iproto_cfg_msg cfg_msg;
 	iproto_cfg_msg_create(&cfg_msg, IPROTO_CFG_LISTEN);
 	cfg_msg.binary = binary;
+#ifndef NDEBUG
+	struct errinj *inj = errinj(ERRINJ_IPROTO_CFG_LISTEN_FAIL, ERRINJ_BOOL);
+	if (inj->bparam) {
+		inj->bparam = false;
+		diag_set(ClientError, ER_INJECTION,
+			 "Iproto listen fail");
+
+		return -1;
+	};
+#endif
```

Comment:
Why do you need this error injection? Can't you pass an invalid URI to listen instead, as you demonstrated in the bug description?

Anyway, if you do need this error injection, it should be added to iproto_do_cfg_f so as to trigger the existing error path, not introduce a new one.

Reply:
Invalid listen uri fails in 'bind' function. I really don't know how to check it  without injection

Reply:
'iproto_do_cfg_f ' work in iproto thread, errinj sets in tx thread. When i done this injection  in iproto thread 1/200 time of test launch is failed!